src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6916,6 +6916,10 @@ final boolean readPacket() throws SQLServerException {
 
             // if messageType is RPC or QUERY, then increment Counter's state
             if (tdsChannel.getWriter().checkIfTdsMessageTypeIsBatchOrRPC() && null != command) {
+                if (null == command.getCounter()) {
```

Comment:
maybe we should add some more logging here to help us debug when this occured?

Reply:
Added a line of logging to let us know when we're supposed to be increasing the counter.